mysql/src/finagle_clojure/mysql/client.clj

```diff
@@ -0,0 +1,351 @@
+
+  *Arguments:*
+
+    * `rs`: a [[com.twitter.finagle.exp.mysql.ResultSet]]
+
+  *Returns:*
+
+    a Clojure vector of maps"
+  (->> (.rows rs)
+       (scala/scala-seq->vec)
+       (map Row->map)))
+
+(defn- param [p]
+  (reify Stack$Param (default [this] p)))
+
+(defn ^Mysql$Client mysql-client
+  "Initialize a configurable MySQL stack client. The `rich-client` function must be called once configured
+  in order to execute live queries against this, like so:
+
+  ```
+  (-> (mysql-client)
+      (with-database \"somedb\")
+      (with-credentials \"test\" \"test\")
+      (rich-client))
+  ```
+
+  *Arguments:*
+
+    * None.
+
+  *Returns:*
+
+    A new [[com.twitter.finagle.exp.Mysql$Client]]."
+  []
+  (Mysql/client))
+
+(defn ^Mysql$Client with-credentials
+  "Configure the given `Mysql.Client` with connection credentials.
+
+  *Arguments:*
+
+    * `client`: a `Mysql.Client`
+    * `user`: a database username
+    * `pwd`: a database password
+
+  *Returns:*
+
+    the given `Mysql.Client`"
+  [^Mysql$Client client user pwd]
+  (.withCredentials client user pwd))
+
+(defn ^Mysql$Client with-database
+  "Configure the given `Mysql.Client` with a database.
+
+  *Arguments:*
+
+    * `client`: a `Mysql.Client`
+    * `db`: the name of a database
+
+  *Returns:*
+
+    the given `Mysql.Client`"
+  [^Mysql$Client client db]
+  (.withDatabase client db))
+
+(defn ^Mysql$Client with-charset
+  "Configure the given `Mysql.Client` with a charset.
+
+  *Arguments:*
+
+    * `client`: a `Mysql.Client`
+    * `charset`: a number representing the charset
+
+  *Returns:*
+
+    the given `Mysql.Client`"
+  [^Mysql$Client client charset]
+  (.withCharset client (short charset)))
+
+(defn ^Mysql$Client configured
+  "Configure the given `Mysql.Client` with an arbitrary `Stack.Param`.
+
+  *Arguments:*
+
+    * `client`: a `Mysql.Client`
+    * `charset`: an arbitrary `Stack.Param`
+
+  *Returns:*
+
+    the given `Mysql.Client`"
+  [^Mysql$Client client stack-param]
+  (.configured client stack-param (param stack-param)))
+
+(defn ^Client rich-client
+  "Converts the given `Mysql.Client` into a rich client, which is used for actually performing queries.
+
+  *Arguments:*
+
+    * `client`: a `Mysql.Client`
+    * `dest`: a string or `Name` of the server location
+
+  *Returns:*
+
+    a new [[com.twitter.finagle.exp.mysql.Client]] used for real queries"
+  ([^Mysql$Client client dest]
+    (.newRichClient client dest))
+  ([^Mysql$Client client dest label]
+    (.newRichClient client dest label)))
+
+(defn ^Future query
+  "Given a rich client and a SQL string, executes the SQL and returns the result as a Future[Result].
+
+  *Arguments:*
+
+    * `client`: a rich MySQL `Client`
+    * `sql`: a SQL string
+
+  *Returns:*
+
+    a `Future` containing a [[com.twitter.finagle.exp.mysql.Result]]"
+  [^Client client sql]
+  (.query client sql))
+
+
+;; TODO Replace this with core version of lift-fn once merged.
```

Comment:
Function lifting has landed on master. you can replace this with `finagle-clojure.scala/lift->fn1`